docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,254 @@
+
+- **`ExtendEtcdSnapshotImmutabilityTask` Specification:**
+
+  - Runs `etcdbrctl renew-snapshot` to extend the immutability of the latest snapshot.
+  - Runs `etcdbrctl garbage-collect --garbage-collection-policy <policy>` to remove old snapshots.
+
+- **EtcdOperatorTask Controller Logic:**
+
+  - The operator-tasks controller reacts to the creation of the custom resource and deploys a job named `<etcd-name>-extend-immutability`, which is the `ExtendEtcdSnapshotImmutabilityTask` job.
+  - The controller reports metrics regarding the `ExtendEtcdSnapshotImmutabilityTask` job, which can be used to raise alerts for operators if immutability has not been extended.
+
+By periodically re-uploading the latest snapshot during hibernation, the authors ensure that the immutability period is extended, and the backups remain **protected throughout the hibernation period**.
+
+## Compatibility
+
+The proposed changes are fully compatible with existing etcd clusters and backup processes.
+
+- **Backward Compatibility:**
+  - Existing clusters without immutable buckets will continue to function without change.
+  - The introduction of the `ExtendEtcdSnapshotImmutabilityTask` does not affect clusters that are not hibernated.
+- **Forward Compatibility:**
+  - Clusters can opt-in to use immutable backups by configuring the bucket accordingly.
+  - The controller's logic to handle hibernation is additive and does not interfere with existing workflows.
+
+## Risks and Mitigations
+
+- **Increased Storage Costs:**
+  - **Risk:** Copying snapshots or frequent snapshots may lead to increased storage usage.
+  - **Mitigation:** Since only the latest full snapshot is copied, the additional storage usage is minimal. Garbage collection helps manage storage utilization.
+
+- **Operational Complexity:**
+  - **Risk:** Introducing new resources and processes might add complexity.
+  - **Mitigation:** The processes are automated within the controller, requiring minimal operator intervention. Clear documentation and tooling support will help manage complexity.
+
+- **Failed Snapshot Before Hibernation:**
+  - **Risk:** Failure to take a full snapshot before hibernation could delay the hibernation process and potentially compromise data integrity.
+    - **Mitigation:** Implement robust error handling and retry mechanisms to ensure snapshots are taken successfully before hibernation. Notify operators of any failures by updating the `etcd.status.lastErrors` and `etcd.status.lastOperation` fields. Additionally, operators can leverage the metrics provided by `ExtendEtcdSnapshotImmutabilityTask`, which follows the [operator task framework](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/05-etcd-operator-tasks.md#metrics), to trigger alerts. This ensures timely intervention and resolution of issues, maintaining the integrity and availability of the etcd cluster state.
```

Comment:
The risk mentioned is failure to take a full snapshot prior to hibernation. But in the mitigation you are also talking about ` ExtendEtcdSnapshotImmutabilityTask` which only comes into play post hibernation which automatically means that the full snapshot was taken successfully, right?